### ISSUE TITLE
Incorrect "Unused LANGUAGE pragma" warning with typed Template Haskell

### ISSUE THREAD

**Issue by ocharles** [ndmitchell/hlint#1531]
The following

```haskell
{-# LANGUAGE TemplateHaskell #-}

module Main (main) where

-- ch-test
import Test.Tasty.Hedgehog.Extra (testGenerator)

-- ch-uuid
import Data.UUID.Gen qualified as Gen

-- tasty
import Test.Tasty (defaultMain, testGroup)


main :: IO ()
main = defaultMain $ testGroup "Data.UUID.Gen" [$$(testGenerator [||Gen.uuid||])]
```

produces 

```
ch-uuid/tests/Main.hs:1:1-32: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TemplateHaskell #-}
Perhaps you should remove it.
Note: may require `{-# LANGUAGE TemplateHaskellQuotes #-}` adding to the top of the file
```

But this is incorrect, as removing `TemplateHaskell` (or changing it to `TemplateHaskellQuotes`) causes the file to fail to compile.

**Comment by nitinprakash96** [comment]
+1. Facing same issue.

```haskell
{-# LANGUAGE TemplateHaskell #-}

module Git where

import qualified GitHash as Git

version :: String
version = Git.giHash $$(Git.tGitInfoCwd)
```

**Comment by jhrcek** [comment]
I looked into this and it seems that isQuasiQuoteSplice, which is used to recognize TH usage here
https://github.com/ndmitchell/hlint/blob/b73274914d2bd3af0e4188a3de29b300631f06a5/src/Hint/Extensions.hs#L428

has changed to be more restrictive in recent versions of ghc-lib-praser:

Before GHC 9.6 it included both typed and untyped TH splices, but in ghc >=9.6 it only includes untyped splices:
https://hackage.haskell.org/package/ghc-lib-parser-ex-9.6.0.1/docs/src/Language.Haskell.GhclibParserEx.GHC.Hs.Expr.html#isQuasiQuoteSplice

Compare:
https://hackage.haskell.org/package/ghc-9.4.4/docs/Language-Haskell-Syntax-Expr.html#t:HsSplice
https://hackage.haskell.org/package/ghc-9.6.1/docs/Language-Haskell-Syntax-Expr.html#t:HsUntypedSplice

Not sure how to fix this yet :thinking:

**Comment by shayne-fletcher** [comment]
not sure this is quite ready to be closed. re-opening while unsure.

**Comment by jhrcek** [comment]
@shayne-fletcher I think it's safe to close this issue. I tested my fix both reproducers and can confirm that the false positive is no longer emitted for either.
